source/infrastructure/lib/core-scheduler.ts

```diff
@@ -0,0 +1,159 @@
+import {LambdaToDynamoDB, LambdaToDynamoDBProps} from "@aws-solutions-constructs/aws-lambda-dynamodb";
+import {Aws, RemovalPolicy, Stack} from "aws-cdk-lib";
+import {NagSuppressions} from "cdk-nag";
+
+import * as iam from "aws-cdk-lib/aws-iam"
+import * as s3 from "aws-cdk-lib/aws-s3"
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as cdk from "aws-cdk-lib";
+import * as dynamodb from "aws-cdk-lib/aws-dynamodb";
+
+export interface InstanceSchedulerLambdaProps {
+
+  readonly solutionVersion: string
+  readonly solutionTradeMarkName: string
+  readonly solutionsBucket?: s3.IBucket
+  readonly memorySize: number
+  readonly schedulerRole: iam.Role
+  /**
+   * Lambda Function environment variables
+   */
+  readonly environment?: {
+    [key: string]: string;
+  }
+}
+export class CoreScheduler {
+
+  public readonly lambdaFunction: lambda.Function;
+  public readonly configTable: dynamodb.Table
+  private readonly stateTable: dynamodb.Table
+  private readonly maintenanceWindowTable: dynamodb.Table
+
+  constructor(scope: Stack, props: InstanceSchedulerLambdaProps) {
+
+    const lambdaToDynamoDbConstruct = new LambdaToDynamoDB(scope, 'instance-scheduler-lambda', this.lambdaToDynamoPropsFrom(props))
+
+    this.lambdaFunction = lambdaToDynamoDbConstruct.lambdaFunction;
+    this.stateTable = lambdaToDynamoDbConstruct.dynamoTable;
+
+    const cfnStateTable = this.stateTable.node.defaultChild as dynamodb.CfnTable
+    cfnStateTable.overrideLogicalId('StateTable')
+    cfnStateTable.addPropertyOverride("SSESpecification", {
+      "KMSMasterKeyId": { "Ref": "InstanceSchedulerEncryptionKey" },
+      "SSEEnabled": true,
+      "SSEType": 'KMS'
+    })
+
+    this.configTable = new dynamodb.Table(scope, 'ConfigTable', {
```

Comment:
The table properties are repeated three times in this file. Can we consolidate them?

Reply:
the tables have subtly different properties and configurations. Perhaps they could be better consolidated in a followup refactor? (perhaps also removing the LambdaToDynamoDB construct like we had discussed this a.m.)